src/Interpreters/TransactionLog.cpp

```diff
@@ -146,8 +143,7 @@ void TransactionLog::loadEntries(Strings::const_iterator beg, Strings::const_ite
     }
     futures.clear();
 
-    NOEXCEPT_SCOPE;
-    LockMemoryExceptionInThread lock_memory_tracker(VariableContext::Global);
+    NOEXCEPT_SCOPE_STRICT;
```

Comment:
I see the difference between them, but can you explain why strict variant is used here?

Reply:
Because initially we had only strict variant and it was the only place where this macro was used. But strict variant is not generic enough to use it in other places. I decided to leave strict variant here because something is probably going wrong if some code tries to initialize transaction log during stack unwinding.